lib/grpc_reflection/service/builder.ex

```diff
@@ -104,10 +120,75 @@ defmodule GrpcReflection.Service.Builder do
       root_symbols = %{symbol => response}
       root_files = %{(symbol <> ".proto") => response}
 
-      {%Agent{files: root_files, symbols: root_symbols}, referenced_types}
+      extension_file = symbol <> "Extension.proto"
+
+      {extension_numbers, extension_payload} =
+        process_extensions(mod, symbol, extension_file, descriptor)
+
+      {root_extensions, root_files} =
+        if extension_numbers != nil and extension_payload != nil do
```

Comment:
minor nit, I prefer using `is_nil`, but I also have a few scars from languages that coerce types; this should be fine

Reply:
these (`extension_numbers` and `extension_payload`) appear to always both be nil, or both hold a value.

I don't have a direct suggestion but it feels like this could be a bit simpler, maybe an `{:ok, {root_extensions, root_files}}` response from `process_extensions` and you check for `:ok` instead of nil twice?  just a guess.